lib/ui/painting/codec.cc

```diff
@@ -363,6 +363,43 @@ void Codec::dispose() {
   ClearDartWrapper();
 }
 
+const SkBitmap* MultiFrameCodec::DecodedFrame::ConsumeCachedBitmap() {
+  timesUsed_++;
+  return bitmap_.get();
+}
+
+bool MultiFrameCodec::DecodedFrame::CachedBitmapExists() {
+  return bitmap_ && bitmap_->getPixels() != nullptr;
+}
+
+void MultiFrameCodec::DecodedFrame::MaybeCache(SkBitmap* bitmap) {
+  timesUsed_ = 0;
+  const size_t cachedFrameSize = bitmap->computeByteSize();
+  // Don't cache this if we already have an entry or if it's non required and
+  // would overflow the cache.
+  if (bitmap_ ||
+      (numDependentFrames_ == 0 && codec_->OverflowsCache(cachedFrameSize))) {
+    return;
+  }
+  bitmap_ = std::make_unique<SkBitmap>(*bitmap);
+  codec_->decodedCacheSize_ += cachedFrameSize;
```

Comment:
If this overflows the cache and we have cached a non-required frame would be nice to evict the non required frame.

(Probably out of scope for this PR, if you think it makes sense a TODO will make it)

Reply:
Added a TODO at the top of this function and in the header for `DecodedFrame`. We'll probably want to follow up on this, should just be aware that at this point we'll start impacting performance and CPU usage in some cases to keep memory usage down.